crates/rspack_swc_visitors/src/define.rs

```diff
@@ -37,10 +36,14 @@ pub fn define(opts: &Define) -> impl Fold {
           None,
           &mut vec![],
         )
-        .unwrap_or_else(|_| panic!("builtins.define: Failed to parse {:?}", target))
       };
 
-      (NodeIgnoringSpan::owned(*target), *replacement)
+      if let (Ok(target), Ok(replacement)) = (target, replacement) {
+        Some((NodeIgnoringSpan::owned(*target), *replacement))
+      } else {
+        eprintln!("builtins.define: Failed to parse {:?}", raw_target);
```

Comment:
This would better be a diagnostic, and more specifically it's a warning since I didn't see any error emitted in `webpack.DefinePlugin`(however we use swc parse in rspack, which does not block the build and you just keep the `target` or `replacement` as-is). Leveraging `eprintln` ends up errors not being showed in the `Stats`.

Reply:
`builtins.define` is also a deprecated option, please use `DefinePlugin` instead.

Reply:
if you set plugin `new rspack.DefinePlugin({ "__DEV__": "😄" })`, then it will cause errors and print logs. 
So according to your idea, should I diagnose it before it compiles and runs?
<img width="316" alt="image" src="https://github.com/web-infra-dev/rspack/assets/165626830/6d1117ba-4e9c-4bae-97f3-31b9883a22c0">

Reply:
Or conduct a diagnosis before it executes `define`?

Reply:
I suggest to pass [`diagnostic`](https://github.com/web-infra-dev/rspack/blob/0e438151e9742e182bb75c9393c4689d890d975e/crates/rspack_plugin_javascript/src/parser_and_generator/mod.rs#L100) to `define`.